File: frontend/src/app/shared/model/Token.ts
export interface Token {
  token: string;
  purpose: string;
  email: string;
  isUsed: boolean;
}

export class TokenConstructor {
  static new(): Token {
    return {
      token: '',
      purpose: '',
      email: null,
      isUsed: false,
    };
  }

  static newFromObj(obj: Token): Token {
    if (obj === null) {
      return null;
    }

    return {
      token: obj.token,
      purpose: obj.purpose,
      email: obj.email,
      isUsed: obj.isUsed,
    };
  }
}

File: backend/src/main/java/blok2/model/Token.java
package blok2.model;

public class Token {

    private String token;
    private String purpose;
    private String email;
    private boolean isUsed;

    public Token() {
    }

    public Token(String token, String purpose, String email, boolean isUsed) {
        this.token = token;
        this.purpose = purpose;
        this.email = email;
        this.isUsed = isUsed;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getPurpose() {
        return purpose;
    }

    public void setPurpose(String purpose) {
        this.purpose = purpose;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public boolean getIsUsed() {
        return isUsed;
    }

    public void setIsUsed(boolean used) {
        isUsed = used;
    }

    @Override
    public String toString() {
        return "Token{" +
                "token='" + token + '\'' +
                ", purpose='" + purpose + '\'' +
                ", email='" + email + '\'' +
                ", isUsed=" + isUsed + '\'' +
                '}';
    }
}
